src/condor_tools/history.cpp

```diff
@@ -1419,5 +1451,112 @@ static int set_print_mask_from_stream(
 	return err;
 }
 
+//Find and add all epoch files in the given epochDirectory then sort
+//based on cluster.proc & backwards/forwards
+static void findEpochFiles(std::deque<std::string> *epochFiles){
+	std::string jobID; //Make jobId to search for .XXX.YY.
+	bool oneJob = false;
+	if (cluster != -1) {
+		if (proc != -1) { formatstr(jobID,".%d.%d.",cluster,proc); oneJob = true; }
+		else { formatstr(jobID,".%d.",cluster); }
+	}
+
+	//Get all the epoch ads we want
+	Directory dir(epochDirectory);
+	std::string file;
+	do {
+		//Get file name in directory
+		const char *curr = dir.Next();
+		file = curr ? curr : "";
+		//Check if filename matched job.runs.X.Y.ads
+		if (starts_with(file,"job.runs.") && ends_with(file,".ads")) {
+			//If no jobID then add all matching epoch ads
+			if (jobID == "") {
+				epochFiles->push_back(file);
+			} else { //Otherwise filter based on if the jobID .XXX.YY. is found in the filename
+				if (file.find(jobID) != std::string::npos) {
+					epochFiles->push_back(file);
+					if (oneJob) return;
+				}
+			}
+		}
+	} while(file != "");
+
+	//If backwards then reverse the data structure
+	if (backwards) { std::reverse(epochFiles->begin(),epochFiles->end()); }
+}
+
+static void readHistoryFromEpochs(bool fileisuserlog, const char *JobHistoryFileName, const char* constraint, ExprTree *constraintExpr){
+	printHeader();
 
+	//If we were passed a specific file name then read that.
+	if (JobHistoryFileName) {
+		if (fileisuserlog) {
```

Comment:
I don't think this is a option?

Reply:
Has been decided to be left in for now and dealt with in later changes regarding updating history file writing code